docs/testing/e2e_test_suite.rst

```diff
@@ -0,0 +1,249 @@
+
+.. code-block:: PHP
+
+    <?php
+
+    class TestSuiteNameCest
+    {
+        public function _before(AcceptanceTester $I)
+        {
+            // Code to run before each test
+        }
+
+        public function _after(AcceptanceTester $I)
+        {
+            // Code to run after each test
+        }
+
+        // Define your test methods
+
+        public function login(AcceptanceTester $I)
+        {
+            $I->amOnPage('/s/login');
+            $I->fillField('#username', $name);
+            $I->fillField('#password', $password);
+            $I->click('button[type=submit]');
+            $I->see('Dashboard');
+        }
+    }
+
+3. Utilize PageObjects and StepObjects:
+
+Organize your tests by using PageObject and StepObject classes. This keeps your tests clean and maintainable by separating locators and test steps into reusable components.
+
+- Generate PageObject with:
+
+.. code-block:: bash
+
+    bin/codecept generate:pageobject acceptance ExamplePage
+
+This will create an ExamplePage.php file in /tests/Support/Page/.
+
+- Generate step objects with:
+
+.. code-block:: bash
+
+    bin/codecept generate:stepobject acceptance Example
+
+This will create an Example.php file in/tests/Support/Step/Acceptance.
+
+Running Tests
+=============
+
+You can start tests using the run command provided by Codeception. Here are different ways to run your tests:
+
+**Run All Tests**
+
+.. code-block:: bash
+
+    bin/codecept run
+
+**Run All Acceptance Tests**
+
+.. code-block:: bash
+
+    bin/codecept run acceptance
+
+**Run a Specific Test File**
+
+If you need to run a specific test file, such as ContactManagementCest, use:
+
+.. code-block:: bash
+
+    bin/codecept run acceptance ContactManagementCest
+
+**Run a Specific Test Scenario**
+
+To execute a specific scenario within a test file, you can specify the test method like this:
+
+.. code-block:: bash
+
+    bin/codecept run acceptance ContactManagementCest:createContactFromForm
+
+View Test Results
+=================
+
+After running the tests, the results will be displayed in the terminal. Additionally, any failures will generate snapshots and HTML reports in the _output directory, which you can use for debugging.
+
+Additional Options
+==================
+
+**Print Steps:**
+
+To see a step-by-step breakdown of the test execution, use:
+
+.. code-block:: bash
+
+    bin/codecept run acceptance ContactManagementCest --steps
+
+**Verbose Output**
+
+For more detailed internal debug information, use:
+
+.. code-block:: bash
+
+    bin/codecept run acceptance ContactManagementCest -vvv
+
+View Tests in the Browser
+=========================
+
+You can watch your tests being executed in an automated browser by visiting the following URL: https://mautic.ddev.site:7900/
+
+noVNC Access:
+
+``Password: secret``
+
+Contributing
+************
+
+Contributions to the test suite are welcome. Please follow the guidelines for submitting pull requests.
```

Comment:
Can we link to this so we aren't leaving people hanging? You can use the existing link reference from docs/links/mautic_tester_docs.py with the :xref: syntax.